esp-radio/src/wifi/mod.rs

```diff
@@ -2807,18 +2813,9 @@ impl WifiController<'_> {
     pub fn scan_with_config_sync(
         &mut self,
         config: ScanConfig<'_>,
-    ) -> Result<alloc::vec::Vec<AccessPointInfo>, WifiError> {
-        self.scan_with_config_sync_max(config, usize::MAX)
-    }
-
-    pub fn scan_with_config_sync_max(
-        &mut self,
-        config: ScanConfig<'_>,
-        max: usize,
     ) -> Result<alloc::vec::Vec<AccessPointInfo>, WifiError> {
         esp_wifi_result!(crate::wifi::wifi_start_scan(true, config))?;
-        let result = self.scan_results(max)?;
-        Ok(result)
+        self.scan_results(config.max.unwrap_or(usize::MAX))
```

Comment:
Perhaps it's worth implementing a `max()` function on `ScanConfig` which does this? We only do this in two places currently, so wasn't sure whether or not to do it.